wled00/udp.cpp

```diff
@@ -160,16 +254,66 @@ void handleNotifications()
   
   //notifier and UDP realtime
   if (!packetSize || packetSize > UDP_IN_MAXSIZE) return;
-  if (!isSupp && notifierUdp.remoteIP() == Network.localIP()) return; //don't process broadcasts we send ourselves
 
-  uint8_t udpIn[packetSize +1];
   uint16_t len;
-  if (isSupp) len = notifier2Udp.read(udpIn, packetSize);
-  else        len =  notifierUdp.read(udpIn, packetSize);
+#ifdef WLED_USE_REALTIME_SMOOTHING
+  uint8_t *udpIn = rts.get_free_buffer();
+#else
+  uint8_t udpIn[packetSize+1];
+#endif
+  if (isSupp) {
+    realtimeIP = notifier2Udp.remoteIP();
+    len = notifier2Udp.read(udpIn, packetSize);
+  } else {
+    realtimeIP = notifierUdp.remoteIP();
+    len = notifierUdp.read(udpIn, packetSize);
+  }
+
+  // Realtime or Notifications on main UDP port
+  if (receiveDirect && !isSupp) { 
+    if (udpIn[0] > 0 && udpIn[0] <= 6) {
+      if (udpIn[1] == 0) {
+	realtimeTimeout = 0;
+	return;
+      } else {
+	realtimeLock(udpIn[1]*1000 + 1, REALTIME_MODE_UDP);
+      }
+      if (realtimeOverride || realtimeIP == Network.localIP())
+	return;
+    
+      do { // Read ALL pending realtime packets!
+	if(udpIn[0] > 0 && udpIn[0] <= 6) {  // just skip non-realtime
+#ifdef WLED_USE_REALTIME_SMOOTHING
+	  if (packetSize > RTS_MAX_SIZE) continue;
+	  if (!rts.add(udpIn, packetSize, now)) { // ring overflowed: remove & show one
+	    RealtimeSmooth::rts_frame* show_now = rts.remove(now); 
+	    setRealtimeFrame(show_now);
+	    strip.show();
+	    rts.add(udpIn, packetSize, now, true); // try again
+	  }
+#else // If not RealtimeSmoothing, just display each packet immediately
+	  setRealtime(udpIn,packetSize); // Show realtime packet immediately
+	  strip.show();
+#endif
+	}
+	packetSize = notifierUdp.parsePacket();
+	if(packetSize) { // A packet is waiting... read it!
+#ifdef WLED_USE_REALTIME_SMOOTHING
+	  udpIn = rts.get_free_buffer();
+	  #if DEBUG_RTS
+	  rts.waiting++;
+	  #endif
+#endif
+	  notifierUdp.read(udpIn, packetSize);
+	}
+      } while (packetSize);
+      return;
+    }
+  }
 
   // WLED nodes info notifications
   if (isSupp && udpIn[0] == 255 && udpIn[1] == 1 && len >= 40) {
-    if (!nodeListEnabled || notifier2Udp.remoteIP() == Network.localIP()) return;
+    if (!nodeListEnabled || realtimeIP == Network.localIP()) return;
```

Comment:
Why this change?
It is inappropriate in this case.

Reply:
If you look [just above](https://github.com/Aircoookie/WLED/blob/1f871690655a280c4a74bc681f7c0cb2d4d79247/wled00/udp.cpp#L265), you can see `realtimeIP` will be set to `notifier2Udp.remoteIP()` if `isSupp` is true.  So these are equivalent.  Since I already tested for "self-sent updates" above in the realtime block, this is just a minor optimization.  But the name is certainly misleading I agree.  Perhaps we should call it `remoteIP` and set it once.  I now see a superfluous setting of it below in the TMP2.net block.  Or for code clarity just revert this change.

Reply:
I agree about variable rename. It has to be clear it is notifier*2*Udp and not notifierUdp.

But still you are not correct about its use. realtimeIP is assigned inside if statements (and remains set that way until a new trigger arrives). Node notifications rely only on 2nd UDP port notifications.

Reply:
The if statement only selects which UDP port (1st or 2nd) is involved, so one branch of it will have reliably executed each time the WLED Node if block is executed.  But for readability, easy to switch.

Reply:
Reviewing the code (for the third time, even though the node broadcast code was coded by me) I can see you are wrong.
Notifier relies only on 2nd UDP port broadcasts and not both. realtimeIP (packets) can be transmitted over both ports and therefore realtimeIP can have value not intended for node broadcasts.
I still disapprove the change.

Reply:
Reverted.